[2025-02-07T21:31:10.546+0900] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-07T21:31:10.552+0900] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: run_four_python_files.run_silver scheduled__2025-02-07T12:30:00+00:00 [queued]>
[2025-02-07T21:31:10.555+0900] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: run_four_python_files.run_silver scheduled__2025-02-07T12:30:00+00:00 [queued]>
[2025-02-07T21:31:10.555+0900] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2025-02-07T21:31:10.561+0900] {taskinstance.py:2889} INFO - Executing <Task(BashOperator): run_silver> on 2025-02-07 12:30:00+00:00
[2025-02-07T21:31:10.568+0900] {logging_mixin.py:190} WARNING - /Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=28509) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-02-07T21:31:10.568+0900] {standard_task_runner.py:72} INFO - Started process 28514 to run task
[2025-02-07T21:31:10.567+0900] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'run_four_python_files', 'run_silver', 'scheduled__2025-02-07T12:30:00+00:00', '--job-id', '18', '--raw', '--subdir', 'DAGS_FOLDER/run_four_files.py', '--cfg-path', '/var/folders/sp/pflbc_w93l76v3kjctmz_4tw0000gn/T/tmpwsnut329']
[2025-02-07T21:31:10.569+0900] {standard_task_runner.py:105} INFO - Job 18: Subtask run_silver
[2025-02-07T21:31:10.590+0900] {task_command.py:467} INFO - Running <TaskInstance: run_four_python_files.run_silver scheduled__2025-02-07T12:30:00+00:00 [running]> on host macbook-pro-3.local
[2025-02-07T21:31:10.622+0900] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='run_four_python_files' AIRFLOW_CTX_TASK_ID='run_silver' AIRFLOW_CTX_EXECUTION_DATE='2025-02-07T12:30:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-02-07T12:30:00+00:00'
[2025-02-07T21:31:10.622+0900] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-07T21:31:10.632+0900] {subprocess.py:78} INFO - Tmp dir root location: /var/folders/sp/pflbc_w93l76v3kjctmz_4tw0000gn/T
[2025-02-07T21:31:10.632+0900] {subprocess.py:88} INFO - Running command: ['/bin/bash', '-c', 'python silver/feature_extraction.py']
[2025-02-07T21:31:10.637+0900] {subprocess.py:99} INFO - Output:
[2025-02-07T21:31:10.902+0900] {subprocess.py:106} INFO - Processing ../bronze/BTC_USDT_15m.csv with prefix '15m' ...
[2025-02-07T21:31:10.903+0900] {subprocess.py:106} INFO - Traceback (most recent call last):
[2025-02-07T21:31:10.903+0900] {subprocess.py:106} INFO -   File "/Users/jingyounglee/Desktop/bitcoin-data/silver/feature_extraction.py", line 69, in <module>
[2025-02-07T21:31:10.903+0900] {subprocess.py:106} INFO -     df_processed = read_and_preprocess(csv_path, prefix)
[2025-02-07T21:31:10.904+0900] {subprocess.py:106} INFO -                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-02-07T21:31:10.904+0900] {subprocess.py:106} INFO -   File "/Users/jingyounglee/Desktop/bitcoin-data/silver/feature_extraction.py", line 20, in read_and_preprocess
[2025-02-07T21:31:10.904+0900] {subprocess.py:106} INFO -     df = pd.read_csv(csv_file, index_col='datetime', parse_dates=True)
[2025-02-07T21:31:10.904+0900] {subprocess.py:106} INFO -          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-02-07T21:31:10.905+0900] {subprocess.py:106} INFO -   File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1026, in read_csv
[2025-02-07T21:31:10.905+0900] {subprocess.py:106} INFO -     return _read(filepath_or_buffer, kwds)
[2025-02-07T21:31:10.905+0900] {subprocess.py:106} INFO -            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-02-07T21:31:10.905+0900] {subprocess.py:106} INFO -   File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 620, in _read
[2025-02-07T21:31:10.905+0900] {subprocess.py:106} INFO -     parser = TextFileReader(filepath_or_buffer, **kwds)
[2025-02-07T21:31:10.906+0900] {subprocess.py:106} INFO -              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-02-07T21:31:10.906+0900] {subprocess.py:106} INFO -   File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1620, in __init__
[2025-02-07T21:31:10.906+0900] {subprocess.py:106} INFO -     self._engine = self._make_engine(f, self.engine)
[2025-02-07T21:31:10.906+0900] {subprocess.py:106} INFO -                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-02-07T21:31:10.906+0900] {subprocess.py:106} INFO -   File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1880, in _make_engine
[2025-02-07T21:31:10.906+0900] {subprocess.py:106} INFO -     self.handles = get_handle(
[2025-02-07T21:31:10.907+0900] {subprocess.py:106} INFO -                    ^^^^^^^^^^^
[2025-02-07T21:31:10.907+0900] {subprocess.py:106} INFO -   File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/pandas/io/common.py", line 873, in get_handle
[2025-02-07T21:31:10.907+0900] {subprocess.py:106} INFO -     handle = open(
[2025-02-07T21:31:10.907+0900] {subprocess.py:106} INFO -              ^^^^^
[2025-02-07T21:31:10.907+0900] {subprocess.py:106} INFO - FileNotFoundError: [Errno 2] No such file or directory: '../bronze/BTC_USDT_15m.csv'
[2025-02-07T21:31:10.937+0900] {subprocess.py:110} INFO - Command exited with return code 1
[2025-02-07T21:31:10.942+0900] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/operators/bash.py", line 276, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-02-07T21:31:10.946+0900] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=run_four_python_files, task_id=run_silver, run_id=scheduled__2025-02-07T12:30:00+00:00, execution_date=20250207T123000, start_date=20250207T123110, end_date=20250207T123110
[2025-02-07T21:31:10.951+0900] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-07T21:31:10.951+0900] {standard_task_runner.py:124} ERROR - Failed to execute job 18 for task run_silver (Bash command failed. The command returned a non-zero exit code 1.; 28514)
Traceback (most recent call last):
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jingyounglee/miniforge3/envs/bitcoin/lib/python3.12/site-packages/airflow/operators/bash.py", line 276, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-02-07T21:31:10.982+0900] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-02-07T21:31:10.991+0900] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-07T21:31:10.993+0900] {local_task_job_runner.py:245} INFO - ::endgroup::
